src/arch_ppc.py

```diff
@@ -0,0 +1,862 @@
+        "fmadds": lambda a: BinaryOp.f32(
+            BinaryOp.f32(a.reg(1), "*", a.reg(2)), "+", a.reg(3)
+        ),
+        "fnmadd": lambda a: UnaryOp(
+            op="-",
+            expr=BinaryOp.f64(BinaryOp.f64(a.reg(1), "*", a.reg(2)), "+", a.reg(3)),
+            type=Type.f64(),
+        ),
+        "fnmadds": lambda a: UnaryOp(
+            op="-",
+            expr=BinaryOp.f32(BinaryOp.f32(a.reg(1), "*", a.reg(2)), "+", a.reg(3)),
+            type=Type.f32(),
+        ),
+        "fmsub": lambda a: BinaryOp.f64(
+            BinaryOp.f64(a.reg(1), "*", a.reg(2)), "-", a.reg(3)
+        ),
+        "fmsubs": lambda a: BinaryOp.f32(
+            BinaryOp.f32(a.reg(1), "*", a.reg(2)), "-", a.reg(3)
+        ),
+        "fnmsub": lambda a: UnaryOp(
+            op="-",
+            expr=BinaryOp.f64(BinaryOp.f64(a.reg(1), "*", a.reg(2)), "-", a.reg(3)),
+            type=Type.f64(),
+        ),
+        "fnmsubs": lambda a: UnaryOp(
+            op="-",
+            expr=BinaryOp.f32(BinaryOp.f32(a.reg(1), "*", a.reg(2)), "-", a.reg(3)),
+            type=Type.f32(),
+        ),
+        # TODO: Detect if we should use fabs or fabsf
+        "fabs": lambda a: fn_op("fabs", [a.reg(1)], Type.floatish()),
+    }
+    instrs_load_update: InstrMap = {
+        "lbau": lambda a: handle_load(a, type=Type.s8()),
+        "lbzu": lambda a: handle_load(a, type=Type.u8()),
+        "lhau": lambda a: handle_load(a, type=Type.s16()),
+        "lhzu": lambda a: handle_load(a, type=Type.u16()),
+        "lwzu": lambda a: handle_load(a, type=Type.reg32(likely_float=False)),
+        "lbaux": lambda a: handle_loadx(a, type=Type.s8()),
+        "lbzux": lambda a: handle_loadx(a, type=Type.u8()),
+        "lhaux": lambda a: handle_loadx(a, type=Type.s16()),
+        "lhzux": lambda a: handle_loadx(a, type=Type.u16()),
+        "lwzux": lambda a: handle_loadx(a, type=Type.reg32(likely_float=False)),
+        "lfsu": lambda a: handle_load(a, type=Type.f32()),
+        "lfdu": lambda a: handle_load(a, type=Type.f64()),
+        "lfsux": lambda a: handle_loadx(a, type=Type.f32()),
+        "lfdux": lambda a: handle_loadx(a, type=Type.f64()),
+    }
+
+    # TODO: Unclear if there will be many instructions like this, or if
+    # there will just be separate dicts for each implicit dest register
+    instrs_implicit_destination: ImplicitInstrMap = {
+        "mtlr": (Register("lr"), lambda a: a.reg(0)),
+        "mtctr": (Register("ctr"), lambda a: a.reg(0)),
+    }
+
+    instrs_ppc_compare: PpcCmpInstrMap = {
+        "cmpw": lambda a, op: BinaryOp.sintptr_cmp(a.reg(0), op, a.reg(1)),
+        "cmpwi": lambda a, op: BinaryOp.sintptr_cmp(a.reg(0), op, a.imm(1)),
+        "cmplw": lambda a, op: BinaryOp.uintptr_cmp(a.reg(0), op, a.reg(1)),
+        "cmplwi": lambda a, op: BinaryOp.uintptr_cmp(a.reg(0), op, a.imm(1)),
+        # TODO: There is a difference in how these two instructions handle NaN
+        # TODO: Assert that the first arg is cr0
```

Comment:
Added the check in. 

This changed 10 of the project tests (out of 1210): on some, it changed the error message, and others now raise `DecompFailure` (because they incorrectly handled instructions before). 

Error message change:
```diff
-src.error.DecompFailure: Instruction argument $r6 must be a literal
+src.error.DecompFailure: Instruction cmpwi $cr6, $r6, 0x0 not supported (first arg must be CR0)
```